packages/node-fetch-server/src/lib/fetch-incoming-message/index.ts

```diff
@@ -0,0 +1,513 @@
+    // The redirect getter steps are to return this’s request’s redirect mode.
+    return this.#state.redirect;
+  }
+
+  /**
+   * Returns request’s sub-resource integrity metadata, which is a
+   * cryptographic hash of the resource being fetched. Its value
+   * consists of multiple hashes separated by whitespace. [SRI]
+   */
+  get integrity() {
+    // The integrity getter steps are to return this’s request’s integrity
+    // metadata.
+    return this.#state.integrity;
+  }
+
+  // Fetch API 'headers' getter
+  get headers() {
+    return this.#headers;
+  }
+
+  /**
+   * Returns a boolean indicating whether or not request can outlive the
+   * global in which it was created.
+   */
+  get keepalive(): boolean {
+    // The keepalive getter steps are to return this’s request’s keepalive.
+    return this.#state.keepalive;
+  }
+
+  /**
+   *  Returns a boolean indicating whether or not request is for a reload
+   *  navigation.
+   */
+  get isReloadNavigation(): boolean {
+    // The isReloadNavigation getter steps are to return true if this’s
+    // request’s reload-navigation flag is set; otherwise false.
+    return this.#state.reloadNavigation;
+  }
+
+  /**
+   * Returns a boolean indicating whether or not request is for a history
+   * navigation (a.k.a. back-forward navigation).
+   */
+  get isHistoryNavigation(): boolean {
+    // The isHistoryNavigation getter steps are to return true if this’s request’s
+    // history-navigation flag is set; otherwise false.
+    return this.#state.historyNavigation;
+  }
+
+  /**
+   * Returns the signal associated with request, which is an AbortSignal
+   * object indicating whether or not request has been aborted, and its
+   * abort event handler.
+   */
+  get signal(): AbortSignal {
+    // The signal getter steps are to return this’s signal.
+    return this.#signal;
+  }
+
+  get body() {
+    if (this.method === 'GET' || this.method === 'HEAD') {
+      return null;
+    }
+
+    // // lazily convert the body to a ReadableStream on first access
+    if (!this.#state.body) {
+      this.#state.body = {
+        stream: Readable.toWeb(this),
+        source: null,
+      };
+    }
+
+    return this.#state.body ? this.#state.body.stream : null;
+  }
+
+  get bodyUsed(): boolean {
+    return !!this.#state.body && isDisturbed(this);
+  }
+
+  get duplex() {
+    return 'half' as const;
+  }
+
+  //#endregion Fetch Getters
+
+  //#region Fetch API
+
+  /**
+   * **DO NOT USE**, this method is not implemented and will always throw. Kept
+   * only for compatibility with type definitions.
+   *
+   * @throws {Error} Method not implemented
+   */
+  // @ts-expect-error
+  public clone(): FetchIncomingMessage {
+    throw new Error('Method not implemented.');
+  }
+
+  /**
+   * **DO NOT USE**, will always throw. Kept only for compatibility with type definitions.
+   *
+   * @deprecated Use `Request.formDataAsync()` instead which returns an `AsyncGenerator<MultipartPart>`
```

Comment:
Where is `Request.formDataAsync` defined? Is that a new spec? Or are you using the form-data-parser package?